### PR TITLE
misc(tooling): Upgrade sbt 0.13 to sbt 1.3.0

### PR DESCRIPTION
**Pull request by xingyif**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [x] Docs have been added / updated (for bug fixes / features) ?

**Related Issue:**
https://github.com/filodb/FiloDB/issues/458

**Changes:**
1. - [x] Introducing build.sbt:
Since FlioBuild is defining the project, it should be organized in build.sbt (according to the new sbt reqs). FiloSettings can be kept in the .scala files because it contains objects that FiloBuild needs.
2. - [x] Some of the common setting things like scalaVersion and organization can move to ThisBuild / x. see https://www.scala-sbt.org/1.x/docs/Using-Sonatype.html#step+4%3A+Configure+build.sbt
3. - [x] Tracking all the dependencies in one file - https://www.scala-sbt.org/1.x/docs/Organizing-Build.html#Tracking+dependencies+in+one+place

**Challenges:**
1.  Got this error `not found: value scalastyleFailOnError; scalastyleFailOnError := true,` in project/FiloSettings.scala at line 61, using sbt 1.3. I went through scalastyle doc http://www.scalastyle.org/sbt.html. Seems like scalastyleFailOnError filed still exists. Currently trying to find out what is wrong.

2. Also in scalastyle doc (linked above): It says that scalastyle >= 0.9.0, should use `testScalastyle := scalastyle.in(Test).toTask("").value`. compare to scalastyle <= 0.8.0, `testScalastyle := org.scalastyle.sbt.ScalastylePlugin.scalastyle.in(Test).toTask("").value`.
I changed the code accordingly. But still got `not found: value scalastyle`.

### CODE REVIEW

**Comment by xingyif** [comment]
Status Update: Fixed sbt compile errors. Tested that the jars still build and able to successfully run kafka end to end testing commands.

Next todo:
Tracking all the dependencies in one file - https://www.scala-sbt.org/1.x/docs/Organizing-Build.html#Tracking+dependencies+in+one+place

**Comment by xingyif** [comment]
@velvia I've upgrade to sbt v1.3.2.
Currently sbt compile & sbt test:compile both ends with success.
I will test tmr to make sure the jars still build and ensure it by doing end-to-end kafka testing.

**Comment by xingyif** [comment]
Testing to make sure the jars still build. Following the steps in Readme.md end-to-end-kafka testing.
1. Was able to build the jars and required objects by running `sbt standalone/assembly cli/assembly gateway/assembly`.
2. Was able to initialize the keysaoces ad tables in Cassandra `./filo-cli -Dconfig.file=conf/timeseries-filodb-server.conf  --command init`
3. Verified `filodb` and `filodb-admin` keyspaces exist.
4. Able to start 2 FiloDB Dev Standalone servers
5. `curl localhost:8080/api/v1/cluster/prometheus/status` returned the following results:
Screen Shot 2019-09-30 at 18 31 28
6. However, when running `./dev-gateway.sh --gen-prom-data conf/timeseries-dev-source.conf`, got the following error:
Screen Shot 2019-09-30 at 18 32 48

**Comment by xingyif** [comment]
> Testing to make sure the jars still build. Following the steps in Readme.md end-to-end-kafka testing.
> 
> 1. Was able to build the jars and required objects by running `sbt standalone/assembly cli/assembly gateway/assembly`.
> 2. Was able to initialize the keysaoces ad tables in Cassandra `./filo-cli -Dconfig.file=conf/timeseries-filodb-server.conf  --command init`
> 3. Verified `filodb` and `filodb-admin` keyspaces exist.
> 4. Able to start 2 FiloDB Dev Standalone servers
> 5. `curl localhost:8080/api/v1/cluster/prometheus/status` returned the following results:
> 
> Screen Shot 2019-09-30 at 18 31 28
> 
> **Note the following error exists when checked out on the `develop` branch:**
> 6. However, when running `./dev-gateway.sh --gen-prom-data conf/timeseries-dev-source.conf`, got the following error:
> Screen Shot 2019-09-30 at 18 32 48
> Cannot find `KAMON counter name=memstore-rows-ingested count=4999` in servers' logs.
> 
> 1. I tried to delete the Kafka topic and re-create it. I was able to delete the topics, but not able to recreate Kafka topics. Ran `kafka-topics --create --zookeeper localhost:2181 --replication-factor 1 --partitions 4 --topic timeseries-dev-ds-1m`. Got
> 
> ```
> Error while executing topic command : Replication factor: 1 larger than available brokers: 0.
> [2019-09-30 19:00:18,778] ERROR org.apache.kafka.common.errors.InvalidReplicationFactorException: Replication factor: 1 larger than available brokers: 0.
>  (kafka.admin.TopicCommand$)
> ```
> 
> 1. Running `./dev-gateway.sh --gen-prom-data` also got:
>    `Error: Could not find or load main class gateway.target.scala-2.11.gateway-0.9-SNAPSHOT`
> 
> Any ideas on what caused this?

**Comment by xingyif** [comment]
Whoops, sorry, accidentally closed the pr. Not intended.

**Comment by velvia** [comment]
@xingyif  you might want to try cleaning out gateway/target/scala-2.11, you might have multiple assembly jars in that folder.

File: project/Dependencies.scala [hunk 1]
```diff
@@ -266,7 +104,7 @@ object FiloBuild extends Build {
     "de.javakaffee"        % "kryo-serializers"      % "0.42" excludeAll(excludeMinlog),
     "io.kamon"             %% "kamon-prometheus"  % "1.1.1",
     // Redirect minlog logs to SLF4J
-     "com.dorkbox"         % "MinLog-SLF4J"       % "1.12",
+    "com.dorkbox"         % "MinLog-SLF4J"       % "1.12",
```
Comment (velvia) [inline project/Dependencies.scala]: realign the % ?

Comment (xingyif) [reply project/Dependencies.scala]: Realighed all %s

**Comment by xingyif** [comment]
@velvia I was able to query FiloDB for the ingested data after cleaning out gateway/target/scala-2.11. 
Screen Shot 2019-10-07 at 18 46 10

Rewrote the commit messages following the Contributing.md guidelines. Anything else to change before we merge this?

**Comment by velvia** [comment]
@xingyif  I restarted the build.  It's hard to see exactly where the failures are as part of the output did not have color.... know why?  Hope this is not an SBT 1.3 thing.... :/

**Comment by xingyif** [comment]
@velvia I looked around and only found https://www.scala-sbt.org/1.x/docs/sbt-1.3-Release-Notes.html#print+command. Doesn't seem related.

I mostly searched `exception` and saw lots of `java.net.ConnectException: Connection refused (Connection refused)` and a few

```
Exception encountered when attempting to run a suite with class name: org.scalatest.DeferredAbortedSuite *** ABORTED ***
[info]   com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'seed-nodes'
[info]   at com.typesafe.config.impl.SimpleConfig.findKeyOrNull(SimpleConfig.java:156)
[info]   at com.typesafe.config.impl.SimpleConfig.findOrNull(SimpleConfig.java:174)
[info]   at com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:188)
[info]   at com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:193)
[info]   at com.typesafe.config.impl.SimpleConfig.getList(SimpleConfig.java:262)
[info]   at net.ceedubs.ficus.readers.CollectionReaders$$anon$1.read(CollectionReaders.scala:14)
[info]   at net.ceedubs.ficus.FicusConfig$class.as(FicusConfig.scala:9)
[info]   at net.ceedubs.ficus.SimpleFicusConfig.as(FicusConfig.scala:16)
[info]   at filodb.coordinator.FilodbSettings$$anonfun$SeedNodes$2.apply(FilodbSettings.scala:31)
[info]   at filodb.coordinator.FilodbSettings$$anonfun$SeedNodes$2.apply(FilodbSettings.scala:31)
[info]   at scala.Option.getOrElse(Option.scala:121)
[info]   at filodb.coordinator.FilodbSettings.SeedNodes$lzycompute(FilodbSettings.scala:31)
[info]   at filodb.coordinator.FilodbSettings.SeedNodes(FilodbSettings.scala:28)
[info]   at filodb.coordinator.FilodbCluster.joinSeedNodes(FilodbCluster.scala:137)
[info]   at filodb.coordinator.ClusterNodeRecoverySpec$$anonfun$4.apply$mcV$sp(FilodbClusterNodeSpec.scala:177)
[info]   at org.scalatest.SuperEngine.registerNestedBranch(Engine.scala:613)
[info]   at org.scalatest.WordSpecLike$class.org$scalatest$WordSpecLike$$registerBranch(WordSpecLike.scala:147)
[info]   at org.scalatest.WordSpecLike$$anon$2.apply(WordSpecLike.scala:880)
[info]   at org.scalatest.words.MustVerb$StringMustWrapperForVerb$class.must(MustVerb.scala:189)
[info]   at org.scalatest.words.MustVerb$$anon$1.must(MustVerb.scala:227)
[info]   at filodb.coordinator.ClusterNodeRecoverySpec.<init>(FilodbClusterNodeSpec.scala:173)
[info]   at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[info]   at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[info]   at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[info]   at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
[info]   at java.base/java.lang.Class.newInstance(Class.java:584)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:468)
[info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:304)
[info]   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[info]   at java.base/java.lang.Thread.run(Thread.java:834)
[2019-10-10 23:09:49,604] ERROR a.r.t.netty.NettyTransport [NettyTransport(akka://akka-test)] - failed to bind to /127.0.0.1:2552, shutting down Netty transport
```

**Comment by xingyif** [comment]
ping @velvia

**Comment by velvia** [comment]
@xingyif say could you help fix this scalastyle error

[error] (query / compileScalastyle) Failing because of negative scalastyle result

```
[info] scalastyle using config /home/travis/build/filodb/FiloDB/scalastyle-config.xml
789[error] /home/travis/build/filodb/FiloDB/query/src/main/scala/filodb/query/exec/BinaryJoinExec.scala:13:0: filodb.query.Query.qLogger is in wrong order relative to filodb.query.exec.binaryOp.BinaryOperatorFunction.
790[info] scalastyle Processed 36 file(s)
791
```

I'm not sure why the test failures.  Will look into them.

**Comment by velvia** [comment]
@xingyif thanks for everything, I have manually updated, fixed build errors, and merged it.

**Pull request closed without merging**